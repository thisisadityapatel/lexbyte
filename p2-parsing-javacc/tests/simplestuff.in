// Test simple terms
// simple_term	::=	IDNUM | IDSET | NUMBER
#thisisanIDNUM;
v_thisisanIDVEC;
1234;

// vec_const	::=	"[" (exp_list)? "]"
[];
[-4+7];
[0,-20];
[#a,2,3];

// Test all declarations
// statement	::=	var_decl | fn_decl
// var_decl	    ::=	VAR ident_list
// ident_list	::=	identifier ("," identifier)*
// identifier	::=	IDNUM | IDBOOL | IDVEC
var v_a2;
var ?B3,#c;
var v_e,#F,?G,?h,v_I,#J;

// fn_decl		::=	FUNCTION identifier "(" (ident_list)? ")" body END
// body		::=	(statement ";")*
function ?A() end;
function v_b(v_c) return 1; end;
function #d(#e,v_F,#G) return 42; end;

