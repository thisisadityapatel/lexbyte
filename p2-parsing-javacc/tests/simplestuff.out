Parsing succeeded.
Parsing succeeded.
Parsing succeeded.
Parsing succeeded.
Parsing succeeded.
Parsing succeeded.
Parsing succeeded.
Parsing succeeded.
Parsing succeeded.
Parsing succeeded.
Parsing succeeded.
Parsing succeeded.
Parsing succeeded.
